pkg/controller/cluster/custom.go

```diff
@@ -0,0 +1,146 @@
+	namespace := tenant.Namespace
+	secretsMap := map[string][]*miniov2.LocalCertificateReference{
+		"Client":   tenant.Spec.ExternalCertSecret,
+		"Minio":    tenant.Spec.ExternalClientCertSecrets,
+		"MinioCAs": tenant.Spec.ExternalCaCertSecret,
+	}
+	var certificates []*miniov2.CustomCertificateConfig
+	var minioExternalServerCertificates []*miniov2.CustomCertificateConfig
+	var minioExternalClientCertificates []*miniov2.CustomCertificateConfig
+	var minioExternalCaCertificates []*miniov2.CustomCertificateConfig
+
+	for certType, secrets := range secretsMap {
+		certificates = nil
+		publicKey := "public.crt"
+		// Iterate over TLS secrets and build array of CertificateInfo structure
+		// that will be used to display information about certs
+		for _, secret := range secrets {
+			keyPair, err := c.kubeClientSet.CoreV1().Secrets(namespace).Get(ctx, secret.Name, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
+			}
+			if v, ok := secretTypePublicKeyNameMap[secret.Type]; ok {
+				publicKey = v
+			}
+			var rawCert []byte
+			if _, ok := keyPair.Data[publicKey]; !ok {
+				return nil, fmt.Errorf("public key: %v not found inside certificate secret %v", publicKey, secret.Name)
+			}
+			rawCert = keyPair.Data[publicKey]
+			var blocks []byte
+			for {
+				var block *pem.Block
+				block, rawCert = pem.Decode(rawCert)
+				if block == nil {
+					break
+				}
+				if block.Type == "CERTIFICATE" {
+					blocks = append(blocks, block.Bytes...)
+				}
+			}
+			// parse all certificates we found on this k8s secret
+			certs, err := x509.ParseCertificates(blocks)
+			if err != nil {
+				return nil, err
+			}
+			for _, cert := range certs {
+				var domains []string
+				if cert.Subject.CommonName != "" {
+					domains = append(domains, cert.Subject.CommonName)
+				}
+				// append certificate domain names
+				if len(cert.DNSNames) > 0 {
+					domains = append(domains, cert.DNSNames...)
+				}
+				// append certificate IPs
+				if len(cert.IPAddresses) > 0 {
+					for _, ip := range cert.IPAddresses {
+						domains = append(domains, ip.String())
+					}
+				}
+				// Register event in case of certificate expiring
+				expiresIn := time.Until(cert.NotAfter)
+				expiresInDays := int64(expiresIn.Hours() / 24)
+				expiresInHours := int64(math.Mod(expiresIn.Hours(), 24))
+				expiresInMinutes := int64(math.Mod(expiresIn.Minutes(), 60))
+				expiresInSeconds := int64(math.Mod(expiresIn.Seconds(), 60))
+				expiresInHuman := fmt.Sprintf("%v days, %v hours, %v minutes, %v seconds", expiresInDays, expiresInHours, expiresInMinutes, expiresInSeconds)
+
+				if expiresIn.Hours()/24 >= 10 && expiresIn.Hours()/24 < 30 {
+					c.RegisterEvent(ctx, tenant, corev1.EventTypeWarning, "CertificateExpiring", fmt.Sprintf("%s certificate '%s' is expiring in %d days", certType, secret.Name, expiresInDays))
+				}
+				if expiresIn.Hours()/24 >= 0 && expiresIn.Hours()/24 < 10 {
```

Comment:
```suggestion
				if expiresInDays >= 0 && expiresInDays < 10 {
```

Reply:
@allanrogerr any reason this was marked as resolved without implementing the suggestion?

Reply:
@anjalshireesh Thanks for the observation. This has been implemented.